.drone.yml

```diff
@@ -313,20 +364,45 @@ steps:
     commands:
       - apk add --no-cache make
       - make helm-package-charts
-
-  - name: "Helm: Publish charts to staging S3"
-    image: plugins/s3
-    settings:
-      bucket:
-        from_secret: HELM_CHART_STAGING_BUCKET
+  - name: Assume AWS Role
+    image: amazon/aws-cli
+    environment:
       access_key:
-        from_secret: PLUGIN_BUILD_USER_STAGING_KEY
       secret_key:
+      AWS_ACCESS_KEY_ID:
+        from_secret: PLUGIN_BUILD_USER_STAGING_KEY
+      AWS_SECRET_ACCESS_KEY:
         from_secret: PLUGIN_BUILD_USER_STAGING_SECRET
-      region: us-west-2
-      source: packages/*
-      target: /
-      strip_prefix: packages
+      AWS_ROLE:
+        from_secret: PLUGIN_BUILD_ROLE_STAGING
+    commands:
+    - aws sts get-caller-identity
+    - SESSION_NAME=$(echo "drone-${DRONE_REPO}/${DRONE_BUILD_NUMBER}" | sed "s|/|-|g")
+    - |
+        printf "[default]\naws_access_key_id = %s\naws_secret_access_key = %s\naws_session_token = %s" \
+          $(aws sts assume-role \
+            --role-arn "$AWS_ROLE" \
+            --role-session-name "$SESSION_NAME" \
+            --query "Credentials.[AccessKeyId,SecretAccessKey,SessionToken]" \
+            --output text) \
+            > /root/.aws/credentials
+    - unset AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY
+    - aws sts get-caller-identity
+    volumes:
+    - name: awsconfig
+      path: /root/.aws
+  - name: "Helm: Publish charts to staging S3"
+    image: amazon/aws-cli
+    environment:
+      AWS_S3_BUCKET:
+        from_secret: HELM_CHART_STAGING_BUCKET
+      AWS_REGION: us-west-2
+    commmands:
+    - cd packages
+    - aws s3 sync . s3://$AWS_S3_BUCKET/
+    volumes:
+    - name: awsconfig
+      path: /root/.aws
 
```

Comment:
```suggestion

volumes:
  - name: awsconfig
    temp: {}
```

Reply:
fixed in 7470b8bd1aababa5041b5fccd487a37484d46f0a where I completed this work